openedx/core/djangoapps/content/block_structure/tests/test_models.py

```diff
@@ -80,7 +79,6 @@ def _verify_update_or_create_call(self, serialized_data, mock_log=None, expect_c
         return bsm
 
     @patch('openedx.core.djangoapps.content.block_structure.models.log')
-    @patch.dict(settings.BLOCK_STRUCTURES_SETTINGS, {'PRUNING_ACTIVE': False})
```

Comment:
@Yagnesh1998 since we now always run the pruner, we need to update this test further.  It calls the `_verify_update_or_create_call` function above and assumes that the last log message will be about the creation or update but when the pruner is active, the last log message is about pruning rather than about whether the BlockStructure was created or updated.

If you run this test locally, you can put a breakpoint after line 74 in this file to see what's happening.  Take a look at the output of `mock_log.info.call_args_list` in the debugger to get a better sense of what's going on.  I think you'll have to update that helper function(`_verify_update_or_create_call`) to look at the last call to `mock_log` if it's a creation but look at the 2nd to last call to `mock_log` if it's an update.

Reply:
@feanil as your seegtion i put brackpoint after 74 line and run test aging using python -m test_models.py and i got this issue any suggestion?
`app@d922e5793b93:~/edx-platform/openedx/core/djangoapps/content/block_structure/tests$ python -m test_models.py 
Traceback (most recent call last):
  File "/opt/pyenv/versions/3.8.15/lib/python3.8/runpy.py", line 185, in _run_module_as_main
    mod_name, mod_spec, code = _get_module_details(mod_name, _Error)
  File "/opt/pyenv/versions/3.8.15/lib/python3.8/runpy.py", line 111, in _get_module_details
    __import__(pkg_name)
  File "/openedx/edx-platform/openedx/core/djangoapps/content/block_structure/tests/test_models.py", line 18, in <module>
    from ..exceptions import BlockStructureNotFound
ImportError: attempted relative import with no known parent package
`